### PR TITLE
added Windows to PR builder pipeline

### PR DESCRIPTION
**Pull request by mihaiplesa**
Closes https://github.com/brave/brave-browser/issues/3251

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [x] Windows
  - [x] macOS
  - [x] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [x] Windows
  - [x] macOS
  - [x] Linux
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Requested a security/privacy review as needed.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -141,58 +219,217 @@ pipeline {
...
+                                sh "npm run create_dist -- ${BUILD_TYPE} --channel=${CHANNEL} --debug_build=false --official_build=true --skip_signing"
+                            }
+                        }
+                        stage("archive") {
+                            steps {
+                                // commented because it takes much longer to copy to Jenkins thant to S3
+                                // archiveArtifacts artifacts: "${OUT_DIR}/unsigned_dmg/*.dmg", fingerprint: true
+                                withAWS(credentials: "mac-build-s3-upload-artifacts", region: "us-west-2") {
+                                    awsIdentity()
+                                    s3Upload(acl: "Private", bucket: "${BRAVE_ARTIFACTS_BUCKET}", includePathPattern: "unsigned_dmg/*.dmg",
```
Comment (mbacchi) [inline Jenkinsfile]: Next step will be to upload the .pkg when release channel.

File: Jenkinsfile [hunk 2]
```diff
@@ -4,134 +4,212 @@ pipeline {
...
                             steps {
-                                sh "npm run create_dist -- Release --channel=${CHANNEL} --debug_build=false --official_build=true"
+                                sh "npm run create_dist -- ${BUILD_TYPE} --channel=${CHANNEL} --debug_build=false --official_build=true"
                             }
-                            post {
-                                always {
-                                    archiveArtifacts artifacts: 'src/out/Release/**/*.deb,src/out/Release/**/*.rpm', fingerprint: true
-                                }
+                        }
+                        stage("archive") {
```
Comment (bsclifton) [inline Jenkinsfile]: this is glorious!

File: Jenkinsfile [hunk 3]
```diff
@@ -141,58 +219,216 @@ pipeline {
...
+                                        xunit([GoogleTest(deleteOutputFiles: true, failIfNotNew: true, pattern: "src/brave_unit_tests.xml", skipNoTestFiles: false, stopProcessingIfError: true)])
+                                    }
+                                    catch (ex) {
+                                        currentBuild.result = "UNSTABLE"
+                                    }
                                 }
                             }
                         }
+                        // TODO: fix running browser tests in non-interactive mode
+                        // stage("test-browser") {
```
Comment (bsclifton) [inline Jenkinsfile]: is there an issue created already for this?

Comment (bsclifton) [reply Jenkinsfile]: NM - just found it! 😄 
https://github.com/brave/devops/issues/573

File: Jenkinsfile (hunk 1 above)
Comment (bsclifton) [reply Jenkinsfile]: @mbacchi is that captured in an issue yet?

**Pull request merged**